server/routes/list.js

```diff
@@ -0,0 +1,20 @@
+const express = require('express');
+const router = express.Router();
+const protect = require('../middleware/auth');
+
+//create
+router.route('/').post(protect, (req,res)=>{res.send()});
+
+//update
+router.route('/').put(protect, (req,res)=>{res.send()});
+
+//delete
+router.route('/').delete(protect, (req,res)=>{res.send()});
+
+//get all
+router.route('/').get(protect,(req,res)=>{res.send()});
+
+//get by id
+router.route('/:listId(\\d+)').get(protect,(req,res)=>{res.send()});
```

Comment:
please, what is the purpose of the (\d+)?

Reply:
so only number can be used as id in routes

Reply:
it would have been relevant if only the ids were numeric, but we use mongodb and the ids are automatically generated as ObjectId in the instance. The ids are in the form of "60b38690772d4e6b782327c0". may be I'm wrong. plz check ...

Reply:
good point on mongodb's keys

Reply:
i didnt know that mongo makes keys like that i will change it